satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -0,0 +1,657 @@
+
+        Files provide angles (solar/satellite zenith & azimuth) at a coarser
+        resolution. Interpolate them to the desired resolution.
+        """
+        angles = self._read_dataset(name)
+        if self._is_high_resol(resolution):
+            target_x = self.nc.coords['x']
+            target_y = self.nc.coords['y']
+        else:
+            target_x = self.nc.coords['x_ir_wv']
+            target_y = self.nc.coords['y_ir_wv']
+        return self._interp_tiepoints(
+            angles,
+            target_x=target_x,
+            target_y=target_y
+        )
+
+    def _get_other_dataset(self, name):
+        """Get other datasets such as uncertainties."""
+        ds = self._read_dataset(name)
+        if name in OTHER_REFLECTANCES:
+            ds = ds * 100  # conversion to percent
+        return ds
+
+    def _get_nc_key(self, ds_name):
+        """Get netCDF variable name for the given dataset."""
+        return self.nc_keys.get(ds_name, ds_name)
+
+    def _read_dataset(self, name):
+        """Read a dataset from the file."""
+        nc_key = self._get_nc_key(name)
+        ds = self.nc[nc_key]
+        if 'y_ir_wv' in ds.dims:
+            ds = ds.rename({'y_ir_wv': 'y', 'x_ir_wv': 'x'})
+        elif 'y_tie' in ds.dims:
+            ds = ds.rename({'y_tie': 'y', 'x_tie': 'x'})
+        elif 'y' in ds.dims and 'y' not in ds.coords:
+            # For some reason xarray doesn't assign coordinates to all
+            # high resolution data variables.
+            ds = ds.assign_coords({'y': self.nc.coords['y'],
+                                   'x': self.nc.coords['x']})
+        return ds
+
+    def _update_attrs(self, ds, info):
+        """Update dataset attributes."""
+        ds.attrs.update(info)
+        ds.attrs.update({'platform': self.filename_info['platform'],
+                         'sensor': self.filename_info['sensor']})
+        ds.attrs['raw_metadata'] = self.nc.attrs
+        ds.attrs['orbital_parameters'] = self._get_orbital_parameters()
+
+    def _calibrate(self, ds, channel, calibration):
+        """Calibrate the given dataset."""
+        ds.attrs.pop('ancillary_variables', None)  # to avoid satpy warnings
+        if channel == 'VIS':
+            return self._calibrate_vis(ds, calibration)
+        elif channel in ['WV', 'IR']:
+            return self._calibrate_ir_wv(ds, channel, calibration)
+        else:
+            raise KeyError('Don\'t know how to calibrate channel {}'.format(
+                channel))
+
+    @abc.abstractmethod
+    def _calibrate_vis(self, ds, calibration):
+        """Calibrate VIS channel. To be implemented by subclasses."""
+        raise NotImplementedError
+
+    def _update_refl_attrs(self, refl):
+        """Update attributes of reflectance datasets."""
+        refl.attrs['sun_earth_distance_correction_applied'] = True
+        refl.attrs['sun_earth_distance_correction_factor'] = self.nc[
+            'distance_sun_earth'].item()
```

Comment:
>  Also, how is it with sun-earth distance in these data?

Calibration to reflectance includes the sun earth distance correction and attributes are updated here.